Logger/BaseLogger.cs

```diff
@@ -1,7 +1,11 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
     public abstract class BaseLogger
     {
+        private string? _className;
+        public string ClassName { get => _className; set => _className = value; }
```

Comment:
While this works, it is not an **auto property** as is required in the assignment prompt. Using an **auto property** would look like this:
```suggestion
        public string ClassName { get; set; }
```
When using **auto-implemented properties** "the compiler creates a private, anonymous backing field that can only be accessed through the property's get and set accessors" [(see docs for more details).](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties) This means that declaring a private backing field and referencing that in your getters and setters is redundant.